vmm/src/api/http.rs

```diff
@@ -170,18 +168,18 @@ lazy_static! {
         r.routes.insert(endpoint!("/vm.resume"), Box::new(VmActionHandler::new(VmAction::Resume)));
         r.routes.insert(endpoint!("/vm.shutdown"), Box::new(VmActionHandler::new(VmAction::Shutdown)));
         r.routes.insert(endpoint!("/vm.reboot"), Box::new(VmActionHandler::new(VmAction::Reboot)));
-        r.routes.insert(endpoint!("/vm.snapshot"), Box::new(VmSnapshot {}));
-        r.routes.insert(endpoint!("/vm.restore"), Box::new(VmRestore {}));
+        r.routes.insert(endpoint!("/vm.snapshot"), Box::new(VmActionHandler::new(VmAction::Snapshot(Arc::default()))));
```

Comment:
I wanted to use `Arc::zeroed()` here to provide the placeholder value but that's nightly only.